test/sanity/validate-modules/main.py

```diff
@@ -1206,6 +1206,13 @@ def _validate_argument_spec(self, docs, spec, kwargs):
                     msg='argument_spec for "%s" defines type="bool" but documentation does not' % (arg,)
                 )
 
+            if 'type' not in data and doc_type == 'bool':
```

Comment:
We probably want to be more generic here.  Now that we are allowing all values in documentation, we shouldn't care strictly about `bool` but just ensure it matches.

Reply:
Make sense to me. I will add other datatypes here as well.